### ISSUE TITLE
Verify response value types

### ISSUE THREAD

**Issue by oqq** [imbo/behat-api-extension#41]
Currently there is no way to test the type from a json value.

Given the following response body
```json
{
        "timed_out": false,
        "amount": 15,
        "max_score": 0.15367,
        "items": []
}
```

Would like it where viable to test 

- `timed_out` to be a `bool`
- `amount` to be a `integer`
- `max_score` to be a `float`
- `items ` to be a `array`

**Comment by christeredvartsen** [comment]
Added to the newly created `v2.0` milestone. When other tasks have been done I can come up with a suggestion to an implementation for this.

First thought is to add some sort of callback (like `@length` for instance) called `@type` or something similar.

**Comment by christeredvartsen** [comment]
Closed by commits from #42.
